go/enclave/rollupchain/rollup_chain.go

```diff
@@ -157,21 +130,32 @@ func (rc *RollupChain) ProduceGenesisRollup(blkHash common.L1RootHash) (*core.Ro
 }
 
 // ProcessL1Block is used to update the enclave with an additional L1 block.
-func (rc *RollupChain) ProcessL1Block(block types.Block, isLatest bool) (*common.BlockSubmissionResponse, error) {
+func (rc *RollupChain) ProcessL1Block(block types.Block, isLatest bool, isSequencer bool) (*common.BlockSubmissionResponse, error) {
 	rc.blockProcessingMutex.Lock()
 	defer rc.blockProcessingMutex.Unlock()
 
+	// We update the L1 chain state.
 	err := rc.insertAndStoreL1Block(block, isLatest)
 	if err != nil {
-		return nil, err
+		return nil, rc.rejectBlockErr(err)
 	}
 
-	l2Head, err := rc.updateHeads(&block)
+	// We update the L1 and L2 chain heads.
+	l2Head, isUpdatedRollupHead, err := rc.updateHeads(&block)
 	if err != nil {
 		return nil, rc.rejectBlockErr(err)
 	}
 
-	return rc.produceBlockSubmissionResponse(&block, l2Head)
+	// If we're the sequencer and we've ingested a rollup, we produce a new one.
+	var rollup *common.ExtRollup
+	if isSequencer && isUpdatedRollupHead {
```

Comment:
This was crucial - we should only produce a rollup once a new one has been ingested.

Obviously, this will have to change in the future, once rollups in L1s are just used to validate the batches we have created. But we're not there yet.